components/strava/actions/list-activities/list-activities.mjs

```diff
@@ -0,0 +1,51 @@
+import strava from "../../strava.app.mjs";
+
+export default {
+  name: "Get Activity List",
+  description: "Returns the activities of an athlete for a specific identifier. [See the docs](https://developers.strava.com/docs/reference/)",
+  key: "strava-get-activity-list",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    strava,
+    before: {
+      type: "integer",
+      label: "Before",
+      description: "An epoch timestamp to use for filtering activities that have taken place before a certain time",
+      optional: true,
+    },
+    after: {
+      type: "integer",
+      label: "After",
+      description: "An epoch timestamp to use for filtering activities that have taken place after a certain time",
+      optional: true,
+    },
+    maxItems: {
+      type: "integer",
+      label: "Max number of activities",
+      description: "Maximum number of activities, if not given all activities are returned",
+      optional: true,
+    },
+  },
+  async run({ $ }) {
+    const activities = [];
+    const resourcesStream = await this.strava.getResourcesStream({
+      resourceFn: this.strava.listActivities,
+      resourceFnArgs: {
+        $,
+        params: {
+          before: this.before,
+          after: this.after,
+          per_page: 100,
+        },
+      },
+      maxItems: this.maxItems,
+    });
+    for await (const activity of resourcesStream) {
+      activities.push(activity);
+    }
+    // eslint-disable-next-line multiline-ternary
+    $.export("$summary", `Successfully fetched ${activities.length} ${activities.length === 1 ? "activity" : "activities"}`);
+    return activities;
+  },
```

Comment:
Just wondering if you have tested this component out? because for what I can see there is no endpoint for `GET /activites` but `/athlete/activities` here is the [reference](https://developers.strava.com/docs/reference/#api-Activities-getLoggedInAthleteActivities)

If `GET /activities` endpoint exists I guess you can omit this comment!

Reply:
Yes it is working. It might be updated somehow. I will check `/athlete/activities` and if it works I will use it, if not I will leave it as `/activities`

Reply:
`/athlete/activities` is also working, since it exists in the docs I changed to `/athlete/activities`